NitroxServer-Subnautica/Resources/Parsers/PrefabPlaceholderGroupsParser.cs

```diff
@@ -132,36 +196,64 @@ private ConcurrentDictionary<string, string[]> GetAllPrefabPlaceholdersGroupsFas
             {
                 AssetTypeValueField monoScript = am.GetBaseField(assetFileInstance, monoScriptInfo);
 
-                if (monoScript["m_Name"].AsString != "PrefabPlaceholdersGroup")
-                {
-                    continue;
-                }
-
-                prefabPlaceholdersGroupHash = new byte[16];
-                for (int i = 0; i < 16; i++)
+                switch (monoScript["m_Name"].AsString)
                 {
-                    prefabPlaceholdersGroupHash[i] = monoScript["m_PropertiesHash"][i].AsByte;
+                    case "SpawnRandom":
+                        spawnRandomHash = new byte[16];
+                        for (int i = 0; i < 16; i++)
+                        {
+                            spawnRandomHash[i] = monoScript["m_PropertiesHash"][i].AsByte;
+                        }
+                        break;
+                    case "PrefabPlaceholdersGroup":
+                        prefabPlaceholdersGroupHash = new byte[16];
+                        for (int i = 0; i < 16; i++)
+                        {
+                            prefabPlaceholdersGroupHash[i] = monoScript["m_PropertiesHash"][i].AsByte;
+                        }
+                        break;
                 }
-
-                break;
             }
 
-            if (prefabPlaceholdersGroupHash.Length != 0)
+            if (prefabPlaceholdersGroupHash.Length > 0 && spawnRandomHash.Length > 0)
             {
                 break;
             }
         }
 
-        Parallel.ForEach(addressableCatalog.Skip(aaIndex), (keyValuePair) =>
+        // Now use the bundle paths and the hashes to find out which items from the catalog are important
+        // We fill prefabPlaceholdersGroupPaths and RandomPossibilitiesByClassId when we find objects with a SpawnRandom
+        Parallel.ForEach(addressableCatalog, (keyValuePair) =>
         {
+            string[] assetPaths = keyValuePair.Value;
+            
             AssetsBundleManager bundleManagerInst = am.Clone();
-            BundleFileInstance bundleFile = bundleManagerInst.LoadBundleFile(bundleManagerInst.CleanBundlePath(keyValuePair.Value[0]));
-            AssetsFileInstance assetFileInstance = bundleManagerInst.LoadAssetsFileFromBundle(bundleFile, 0);
+            AssetsFileInstance assetFileInstance = bundleManagerInst.LoadBundleWithDependencies(assetPaths);
 
             if (assetFileInstance.file.Metadata.TypeTreeTypes.Any(typeTree => typeTree.TypeId == (int)AssetClassID.MonoBehaviour && typeTree.TypeHash.data.SequenceEqual(prefabPlaceholdersGroupHash)))
             {
                 prefabPlaceholdersGroupPaths.TryAdd(keyValuePair.Key, keyValuePair.Value);
             }
+            else if (assetFileInstance.file.Metadata.TypeTreeTypes.Any(typeTree => typeTree.TypeId == (int)AssetClassID.MonoBehaviour && typeTree.TypeHash.data.SequenceEqual(spawnRandomHash)))
```

Comment:
Should wrap in if that checks `assetFileInstance.file.Metadata.TypeTreeTypes.Any(typeTree => typeTree.TypeId == (int)AssetClassID.MonoBehaviour` as both `if` and `else if` check for it.